core/matrix/csr_kernels.hpp

```diff
@@ -185,6 +185,16 @@ namespace kernels {
                           const matrix::Csr<ValueType, IndexType> *orig, \
                           matrix::Diagonal<ValueType> *diag)
 
+#define GKO_DECLARE_CSR_SCALE_KERNEL(ValueType, IndexType)  \
+    void scale(std::shared_ptr<const DefaultExecutor> exec, \
+               const matrix::Dense<ValueType> *alpha,       \
+               matrix::Csr<ValueType, IndexType> *to_scale)
+
+#define GKO_DECLARE_CSR_INV_SCALE_KERNEL(ValueType, IndexType)  \
+    void inv_scale(std::shared_ptr<const DefaultExecutor> exec, \
```

Comment:
Do we really need a separate `inv_scale` operation? Isn't it easily done at the calling site by storing the inverse in another `LinOp` and calling `scale`? Or do you need both operations in the same code path? If that is the case, then this makes sense because we don't want to send another number over PCIe or something just for that. If you do not need both operations in the same run, I'd prefer removing `inv_scale` for now and only adding it when the need arises.

Reply:
If consider dense functionalities, which contains the opposite operation, it might be okay at this moment.
it will depends on how the scalar generate, if it is always generated by user (`alpha = initialize ...`), it should be fine with only one scale function. otherwise, the scale will requires additional inverse kernel to behave the inv_scale